[css-values] Fix CSSAttrValue Fallback comparation

AlexsanderDamaceno opened the pull request:
#### ffa340ae3dee5148462db2ec0e2b8fc00a981bb0
<pre>
[css-values] Fix CSSAttrValue Fallback comparation
<a href="https://bugs.webkit.org/show_bug.cgi?id=282316">https://bugs.webkit.org/show_bug.cgi?id=282316</a>

Reviewed by NOBODY (OOPS!).

When comparing two CSSAttrValue instances in the equals function for example, attr(foo) and attr(foo, &quot;fallback&quot;), the result incorrectly returns true.
This happens because the check was only validating cases where both instances have fallback values
but missed handling the scenario where only one instance has a fallback value.

* Source/WebCore/css/CSSAttrValue.cpp:
(WebCore::CSSAttrValue::equals const):
</pre>

nt1m reviewed (approved):
Not sure we can easily test this, but would be nice to do so if possible.

Thanks for fixing

AlexsanderDamaceno committed:
[css-values] Fix CSSAttrValue Fallback comparation
https://bugs.webkit.org/show_bug.cgi?id=282316

Reviewed by Tim Nguyen.

When comparing two CSSAttrValue instances in the equals function for example, attr(foo) and attr(foo, "fallback"), the result incorrectly returns true.
This happens because the check was only validating cases where both instances have fallback values
but missed handling the scenario where only one instance has a fallback value.

* Source/WebCore/css/CSSAttrValue.cpp:
(WebCore::CSSAttrValue::equals const):

Canonical link: https://commits.webkit.org/286063@main
